src/main/java/step3/domain/Cars.java

```diff
@@ -0,0 +1,28 @@
+package step3.domain;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Cars {
+  private final List<Car> carList;
+
+  public Cars(int number) {
+    this.carList = Stream.generate(Car::new)
+        .limit(number)
+        .collect(Collectors.toList());
+  }
+
+  public void race() {
```

Comment:
Cars 도메인 클래스에 대한 테스트가 없네요
랜덤으로 이동하는 부분을 테스트 코드로 작성하기 어려웠을 것 같은데요
다음과 같이 코드를 짜면 테스트 가능할거에요

```suggestion
  // RacingRule는 interface
  public void race(RacingRule racingRule) {
    carList.forEach(car -> {
      if (racingRule.canMove()) {
        car.move();
      }
    });
  }
```

RacingRule를 인터페이스로 바꾸면 구현 클래스가 필요하겠죠
RandomRacingRule 이라는 클래스와 테스트에 필요한 TestRacingRule 두 클래스를 추가하면 Cars 테스트가 가능합니다
물론 TestRacingRule는 test 패키지에만 구현해야될거에요